helm-charts/dnsbl-exporter/values.yaml

```diff
@@ -0,0 +1,339 @@
+##
+readinessProbe:
+  enabled: true
+  initialDelaySeconds: 30
+  periodSeconds: 5
+  timeoutSeconds: 3
+  failureThreshold: 6
+  successThreshold: 1
+## @param customLivenessProbe Custom livenessProbe that overrides the default one
+##
+customLivenessProbe: {}
+## @param customReadinessProbe Custom readinessProbe that overrides the default one
+#
+customReadinessProbe: {}
+## Custom startupProbe block
+customStartupProbe: {}
+
+image:
+  registry: ghcr.io
+  repository: luzilla/dnsbl_exporter
+  tag: v0.6.0
+  pullPolicy: IfNotPresent
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ## e.g:
+  ## pullSecrets:
+  ##   - myRegistryKeySecretName
+  ##
+  pullSecrets: []
+
+configuration:
+  rbls:
+    # 0SPAM
+    - bl.0spam.org
+    - rbl.0spam.org
+    # NIXSPAM
+    - ix.dnsbl.manitu.net
+    # Spamhaus Zen
+    - zen.spamhaus.org
+  targets:
+    - smtp.fastmail.com
+  resolver: "1.1.1.1:53"
+
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # Labels to add to the service account
+  labels: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+## Application Pod Disruption Budget configuration
+## ref: https://kubernetes.io/docs/tasks/run-application/configure-pdb/
+## @param pdb.create Enable a Pod Disruption Budget creation
+## @param pdb.minAvailable Minimum number/percentage of pods that should remain scheduled
+## @param pdb.maxUnavailable Maximum number/percentage of pods that may be made unavailable
+##
+pdb:
+  create: false
+  minAvailable: 1
+  maxUnavailable: ""
+## Configure Pods Security Context
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+## @param podSecurityContext.enabled Enabled Ghost pods' Security Context
+## @param podSecurityContext.fsGroup Set Ghost pod's Security Context fsGroup
+##
+podSecurityContext:
+  enabled: true
+  # Volumes that support ownership management are modified to be owned and writable by the GID specified in fsGroup
+  fsGroup: 101
+## Configure Container Security Context (only main container)
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container
+## @param containerSecurityContext.enabled Enabled Ghost containers' Security Context
+## @param containerSecurityContext.runAsUser Set Ghost container's Security Context runAsUser
+## @param containerSecurityContext.runAsNonRoot Set Ghost container's Security Context runAsNonRoot
+##
+containerSecurityContext:
+  enabled: true
+  runAsUser: 100
```

Comment:
This is a really good idea. 👍🏼 I think the exporter's image should create a user and run as that (vs. root).

Reply:
https://github.com/Luzilla/dnsbl_exporter/issues/185

Reply:
In my current cluster I run the container with user's ID of 100. The dockerfile looks like this:
```Dockerfile
FROM bitnami/minideb:bookworm

ARG TARGETPLATFORM
ARG BUILDPLATFORM
ARG TARGETOS
ARG TARGETARCH
ARG DNSBL_EXPORTER_VERSION=0.6.0

LABEL author="Andrii Miroshnychenko" \
      maintainer="andrey@hq.resecyrity.com" \
      description="The image which hosts dnsbl-exporter" \
      version=${DNSBL_EXPORTER_VERSION} \
      io.wifimap.vendor="Resecurity, LLC"

ENV DNSBL_EXP_RESOLVER=ubound:53
ENV DNSBL_EXP_RBLS=/etc/dnsbl-exporter/rbls.ini
ENV DNSBL_EXP_TARGETS=/etc/dnsbl-exporter/targets.ini

WORKDIR /usr/bin/

RUN install_packages curl ca-certificates && \
    mkdir -p /etc/dnsbl-exporter && \
    curl --create-dirs -fsSL https://github.com/Luzilla/dnsbl_exporter/releases/download/v${DNSBL_EXPORTER_VERSION}/dnsbl_exporter_${DNSBL_EXPORTER_VERSION}_linux_${TARGETARCH}.tar.gz | tar -xvz && \
    echo "Done in ${TARGETARCH}."

COPY rbls.ini /etc/dnsbl-exporter/
COPY targets.ini /etc/dnsbl-exporter/

EXPOSE 9211

ENTRYPOINT ["/usr/bin/dnsbl-exporter"]
```

`rbls.ini` is actually [this](https://github.com/matteocorti/check_rbl/blob/master/check_rbl.ini). `targets.ini` is something like this:
```ini
[targets]
server=does-not-exist.fqdn
server=smtp.gmail.com
server=mail.gmx.net
server=smtp.yahoo.com
server=smtp.fastmail.com
```

Helm chart actually creates a config map which is mounted into `/etc/dnsbl-exporter`.

Reply:
The importer runs as nobody, see dockerfile for ID/GID.